src/main/java/org/broadinstitute/dsm/jobs/PubSubLookUpJob.java

```diff
@@ -0,0 +1,92 @@
+package org.broadinstitute.dsm.jobs;
+
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.gson.Gson;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.SimpleResult;
+import org.broadinstitute.dsm.model.birch.TestBostonResult;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+public class PubSubLookUpJob implements Job {
+    private static final Logger logger = LoggerFactory.getLogger(PubSubLookUpJob.class);
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        String projectId = "hl7-dev-274911";
+        String subscriptionId = "pegah-results-sub";
+
+        try{
+            ProjectSubscriptionName subscriptionName =
+                    ProjectSubscriptionName.of(projectId, subscriptionId);
+
+            // Instantiate an asynchronous message receiver.
+            MessageReceiver receiver =
+                    (PubsubMessage message, AckReplyConsumer consumer) -> {
+                        // Handle incoming message, then ack the received message.
+                        System.out.println("Id: " + message.getMessageId());
+                        System.out.println("Data: " + message.getData().toStringUtf8());
+                        writeResultsIntoDB(message);
+                        consumer.ack();
+                    };
+
+            Subscriber subscriber = null;
+            try {
+                subscriber = Subscriber.newBuilder(subscriptionName, receiver).build();
+                // Start the subscriber.
+                subscriber.startAsync().awaitRunning();
+                logger.info("Listening for messages on %s:\n", subscriptionName.toString());
+                // Allow the subscriber to run for 30s unless an unrecoverable error occurs.
+                subscriber.awaitTerminated(1L, TimeUnit.MINUTES);
```

Comment:
Interesting. Since we haven't called stop before this line, this trick is so to timeout the subscriber after a minute?

Reply:
based on GCP's guide I think this should do the trick.